tests.py

```diff
@@ -20,3 +24,68 @@ def test_basic(data):
     input_stream = BytesIO(data)
     output_stream = GZIPCompressedStream(input_stream, compression_level=5)
     assert decompress(output_stream.read()) == data
+
+
+class FakeAsyncReader(BaseAsyncIteratorReader):
+    def __init__(self, filename: Path):
+        self._fp = open(str(filename), "rb")
+        self._lock = asyncio.Lock()
+
+    async def read(self, size: int = BUFFER_SIZE):
+        async with self._lock:
+            return self._fp.read(size)
+
+    def __del__(self):
+        self._fp.close()
+
+
+@pytest.mark.parametrize(
+    "expected",
+    [
+        '', 't', 'test',
+        Faker().text(4 * 1024),
+        Faker().text(256 * 1024)
+    ],
+    ids=['0 bytes', '1 bytes', '4 bytes',
+         'fake text - ~4 KB', 'fake text - ~256 KB']
+)
+async def test_gzip_aiter_async_reader(expected, tmpdir):
+    tmp_file = tmpdir / "temp.txt"
```

Comment:
Please, use single quotation through all your changes (unless the use is justified), since the rest of the code use single quotation.
[PEP8](https://www.python.org/dev/peps/pep-0008/#string-quotes): "Pick a rule and stick to it".